ethers-etherscan/src/gas.rs

```diff
@@ -1,27 +1,57 @@
 use crate::{Client, EtherscanError, Response, Result};
-use ethers_core::types::U256;
+use ethers_core::{types::U256, utils::parse_units};
 use serde::{de, Deserialize, Deserializer};
 use std::{collections::HashMap, str::FromStr};
 
+const WEI_PER_GWEI: u64 = 1_000_000_000;
+
 #[derive(Deserialize, Clone, Debug)]
 #[serde(rename_all = "PascalCase")]
 pub struct GasOracle {
-    #[serde(deserialize_with = "deserialize_number_from_string")]
-    pub safe_gas_price: f64,
-    #[serde(deserialize_with = "deserialize_number_from_string")]
-    pub propose_gas_price: f64,
-    #[serde(deserialize_with = "deserialize_number_from_string")]
-    pub fast_gas_price: f64,
+    /// Safe Gas Price in wei
+    #[serde(deserialize_with = "deser_gwei_amount")]
+    pub safe_gas_price: U256,
+    /// Propose Gas Price in wei
+    #[serde(deserialize_with = "deser_gwei_amount")]
+    pub propose_gas_price: U256,
+    /// Fast Gas Price in wei
+    #[serde(deserialize_with = "deser_gwei_amount")]
+    pub fast_gas_price: U256,
+    /// Last Block
     #[serde(deserialize_with = "deserialize_number_from_string")]
     pub last_block: u64,
-    #[serde(deserialize_with = "deserialize_number_from_string")]
+    /// Suggested Base Fee in wei
+    #[serde(deserialize_with = "deser_gwei_amount")]
     #[serde(rename = "suggestBaseFee")]
-    pub suggested_base_fee: f64,
+    pub suggested_base_fee: U256,
+    /// Gas Used Ratio
     #[serde(deserialize_with = "deserialize_f64_vec")]
     #[serde(rename = "gasUsedRatio")]
     pub gas_used_ratio: Vec<f64>,
 }
 
+// This function is used to deserialize a string or number into a U256 with an
+// amount of gwei. If the contents is a number, deserialize it. If the contents
+// is a string, attempt to deser as first a decimal f64 then a decimal U256.
+fn deser_gwei_amount<'de, D>(deserializer: D) -> Result<U256, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    #[derive(Deserialize)]
+    #[serde(untagged)]
+    enum StringOrInt {
+        Number(u64),
+        String(String),
+    }
+
+    match StringOrInt::deserialize(deserializer)? {
+        StringOrInt::Number(i) => Ok(U256::from(i) * WEI_PER_GWEI),
```

Comment:
```suggestion
        StringOrInt::Number(i) => Ok(U256::from(i) * super::GWEI_TO_WEI),
```

Reply:
doesn't exist in this package, that's in the middleware, not in the etherscan

Reply:
Ah I see, gotcha!